db/version_builder.cc

```diff
@@ -809,6 +811,19 @@ class VersionBuilder::Rep {
     return Status::OK();
   }
 
+  Status ApplyCompactCursors(int level,
+                             const InternalKey& smallest_uncompacted_key) {
+    if (level < 0 || level >= num_levels_) {
```

Comment:
We should test shrinking num_levels_ still works.

Reply:
Try this sequence of commands with db_bench:

```
$ TEST_TMPDIR=/dev/shm ./db_bench -benchmarks=filluniquerandom,waitforcompaction,deleteseq,compact -num=100000 -write_buffer_size=524288 -max_bytes_for_level_base=2097152 -max_bytes_for_level_multiplier=2 -target_file_size_base=524288 -compression_type=none -compaction_pri=4 -num_levels=5
...
$ TEST_TMPDIR=/dev/shm ./db_bench -benchmarks=overwrite -use_existing_db=true -num=100000 -write_buffer_size=524288 -max_bytes_for_level_base=2097152 -max_bytes_for_level_multiplier=2 -target_file_size_base=524288 -compression_type=none -compaction_pri=4 -num_levels=2
...
open error: Corruption: VersionBuilder: Cannot add compact cursor (2, in file /dev/shm/dbbench/MANIFEST-000005
```

Reply:
For this case, can I return `Corruption` only when `level < 0` and ignore the case `level >= num_level_`? By "ignore", I mean do not even populate it to `updated_compact_cursors` when `level >= num_level_`.

Reply:
Yes, I think that should work.